pkg/ingestor/parser/internal/mock/types.go

```diff
@@ -0,0 +1,95 @@
+// Code generated by MockGen. DO NOT EDIT.
+// Source: /Users/nathannaveen/go/src/github.com/nathannaveen/guac/pkg/ingestor/parser/common/types.go
+
+// Package mocks is a generated GoMock package.
+package mocks
```

Comment:
does this come with a configuration file? want to ensure that this gets updated when it needs to be. Should we add a make command for this? 

Also a broader question, on testing consistency, i don't think it matters too much now that we have tests that use the declaration type tests definitions vs the just code that does comparison (like current other tests). I do worry a bit on consistency and maintainability. Thoughts @pxp928 @mihaimaruseac @mihaimaruseac @jeffmendoza

Reply:
+1 to adding to Makefile and [CI](https://github.com/guacsec/guac/blob/131eeab8164633412cf15109e7d98fab5ce46dc3/.github/workflows/ci.yaml#L96-L97) to make sure it is always regenerated as needed.

Regarding test suites: I'd recommend striving for consistency, as that makes it easier to transfer knowledge from one area of GUAC to another.

Reply:
What do you mean when you say the following? Could you please explain?
> Also a broader question, on testing consistency, i don't think it matters too much now that we have tests that use the declaration type tests definitions vs the just code that does comparison (like current other tests). I do worry a bit on consistency and maintainability. Thoughts @pxp928 @mihaimaruseac @mihaimaruseac @jeffmendoza

Reply:
I will add this to the CI and makefile.

Reply:
I have updated the Makefile, but didn't need to update the CI because I added the mockgen to the generate which is already being called in the CI.

Reply:
> What do you mean when you say the following? Could you please explain?

So the other tests [collector](https://github.com/guacsec/guac/blob/main/pkg/handler/collector/collector_test.go) and [processor](https://github.com/guacsec/guac/blob/main/pkg/handler/processor/process/process_test.go) all do some type of comparison to check that it received the proper documents/artifacts that the results should be as part of the unit tests. This is a deviation from those so the question comes to whether we want to keep them all consistent such that it's easier to understand/maintain (and for doing knowledge transfer in the future).

Reply:
Thanks for the explanation, Parth! 

So, what is the outcome? Should I rewrite my tests in a similar way to the standard?

Reply:
Ok, sure, I will add the "comparison" checks for the `Test_ParserSubscribe` in the next PR.